src/event.rs

```diff
@@ -74,9 +74,94 @@ pub enum Event<'a, T: 'static> {
     UserEvent(T),
 
     /// Emitted when the application has been suspended.
+    ///
+    /// # Portability
+    ///
+    /// Not all platforms support the notion of suspending applications, and there may be no
+    /// technical way to guarantee being able to emit a `Suspended` event if the OS has
+    /// no formal application lifecycle (currently only Android and iOS do). For this reason
+    /// Winit does not currently try to emit pseudo `Suspended` events before the application
+    /// quits on platforms without an application lifecycle.
+    ///
+    /// In case of corner cases with how platform-specific events are mapped to Winit `Suspended`
+    /// and `Resumed` events it's recommended that applications be able to gracefully handle redundant
+    /// (i.e. back to back) `Suspended` or `Resumed` events.
+    ///
+    /// Also see [`Resumed`][Self::Resumed] notes.
+    ///
+    /// ## Android
+    ///
+    /// On Android specifically, the `Suspended` event relates to two things:
+    /// 1. The application is paused (according to the [Activity lifecycle])
+    /// 2. Or the application has had it's associated SurfaceView destroyed (which is expected
+    /// to closely correlate with being paused)
+    ///
+    /// Applications that need to run on Android should assume their SurfaceView has been
+    /// destroyed, which indirectly invalidates any existing render surfaces that may have been
+    /// created outside of Winit (such as an `EGLSurface`, [`VkSurfaceKHR`] or [`wgpu::Surface`]).
+    ///
+    /// After being `Suspended` on Android applications must drop all render surfaces before
+    /// the event callback completes, which may be re-created when the application is next `Resumed`.
```

Comment:
This seems to repeat, but also slightly contradict the above paragraph. Has my render surface "already been destroyed/invalidated", or will that happen _after completing the event callback_? Perhaps merge the two and make it clear that it is the second (as long as the lock is held, see my other PR).